core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -531,24 +529,26 @@ private fun keepBrakingCurveUnderOverlay(
         // Slice envelope to remove the braking curve part which is after the overlay.
         brakingCurve = Envelope.make(*brakingCurve.slice(brakingCurve.beginPos, overlay.endPos))
     }
-    if (
-        brakingCurve.minSpeed >
-            Envelope.make(
-                    *overlay.slice(
-                        max(brakingCurve.beginPos, beginPos),
-                        min(brakingCurve.endPos, overlay.endPos)
-                    )
-                )
-                .maxSpeed
-    ) {
-        // The full braking curve is above the overlay envelope: nothing to do here.
-        return null
-    }
 
     val points = brakingCurve.iteratePoints().distinct()
     val positions = points.map { it.position }
     val speeds = points.map { it.speed }
     val timeDeltas = brakingCurve.flatMap { it.getTimeDeltas() }
+    val isAboveOverlay =
+        // If the last point is strictly above the overlay, the lowest part of the envelope is above
+        // as well.
+        brakingCurve.endSpeed > overlay.interpolateSpeedLeftDir(brakingCurve.endPos, 1.0) ||
+            // If the last point is on the overlay, if the second to last is above the overlay, then
+            // the lowest part of the envelope is above as well.
```

Comment:
To avoid confusion between "overlay envelope" and "braking curve envelope":
```suggestion
        // If the last point is strictly above the overlay, the lowest part of the braking curve is above
        // as well.
        brakingCurve.endSpeed > overlay.interpolateSpeedLeftDir(brakingCurve.endPos, 1.0) ||
            // If the last point is on the overlay, if the second to last is above the overlay, then
            // the lowest part of the braking curve is above as well.
```